### PR TITLE
Provide keepAlive options for the aggregation server

### LINKED ISSUE #34: Aggregation server needs to actively close connections after they are closed client-side
cc @cw9 

It appears in some cases after the client closed the connections, the servers are unaware that the connections had been closed and still wait on the connection to read more data. This means the server keeps the dead connections around and drives up the goroutine count and memory usage artificially.

### PR DESCRIPTION
**Pull request by xichen2020**
cc @cw9 @prateek @jeromefroe 

This PR adds keepAlive options for the aggregation server to actively close out connections that have been closed by the client.

Fixes #34 .

### CODE REVIEW

File: server/msgpack/options.go [hunk 1]
```diff
@@ -34,6 +41,18 @@ type Options interface {
 	// InstrumentOptions returns the instrument options
 	InstrumentOptions() instrument.Options
 
+	// SetKeepAliveEnabled sets the keep alive state for tcp connections.
+	SetKeepAliveEnabled(value bool) Options
```
Comment (cw9) [inline server/msgpack/options.go]: Can we add Get/Set functions for server.Options instead?

Comment (xichen2020) [inline server/msgpack/options.go]: Sure

File: services/m3aggregator/config/server.go [hunk 2]
```diff
@@ -49,9 +56,17 @@ func (c *MsgpackServerConfiguration) NewMsgpackServerOptions(
 ) msgpack.Options {
 	opts := msgpack.NewOptions().SetInstrumentOptions(instrumentOpts)
 
-	// Set retry options.
-	retryOpts := c.Retry.NewOptions(instrumentOpts.MetricsScope())
-	opts = opts.SetRetryOptions(retryOpts)
+	// Set server options.
+	serverOpts := xserver.NewOptions().
+		SetInstrumentOptions(instrumentOpts).
```
Comment (cw9) [inline services/m3aggregator/config/server.go]: also do Tagged(map[string]string{"server": "msgpack"})?

Comment (xichen2020) [inline services/m3aggregator/config/server.go]: The metric scoped passed in already has `msgpack-server` as part of its name, don't think we need an additional tag here.

Comment (cw9) [inline services/m3aggregator/config/server.go]: Having a tag makes the metrics a lot easier to locate, say we can do service: xxx server: msgpack to easily find all the related metrics.

Comment (xichen2020) [inline services/m3aggregator/config/server.go]: Okay sure

**Pull request merged**